openedx/core/djangoapps/credit/models.py

```diff
@@ -186,8 +196,141 @@ class CreditEligibility(TimeStampedModel):
+        * "approved": The user's credit request has been approved by the provider.
+        * "rejected": The user's credit request has been rejected by the provider.
+
         """
-        unique_together = ('username', 'course')
+        try:
+            return self.statuses.latest().status  # pylint: disable=no-member
+        except CreditRequestStatus.DoesNotExist:
+            return "pending"
+
+    @classmethod
+    def credit_requests_for_user(cls, username):
+        """
+        Retrieve all credit requests for a user.
+
+        Arguments:
+            username (unicode): The username of the user.
+
+        Returns: list
+
+        Example Usage:
+        >>> CreditRequest.credit_requests_for_user("bob")
+        [
+            {
+                "uuid": "557168d0f7664fe59097106c67c3f847",
+                "timestamp": "2015-05-04T20:57:57.987119+00:00",
+                "course_key": "course-v1:HogwartsX+Potions101+1T2015",
+                "provider": {
+                    "id": "HogwartsX",
+                    "display_name": "Hogwarts School of Witchcraft and Wizardry",
+                },
+                "status": "pending"  # or "approved" or "rejected"
+            }
+        ]
+
+        """
+        # To minimize the number of database queries, we execute
+        # the raw SQL query instead of using Django's ORM.
+        # This allows us to use a subquery to
+        # retrieve the most recent status for each credit request.
+        cursor = connection.cursor()
+        cursor.execute(
+            (
+                'SELECT cr.uuid, cr.created, cc.course_key, cp.provider_id, cp.display_name, '
+                'IFNULL(('
+                '    SELECT status FROM credit_creditrequeststatus AS crs '
+                '    WHERE crs.request_id=cr.id '
+                '    ORDER BY created DESC LIMIT 1'
+                '), "pending") '
+                'FROM credit_creditrequest as cr '
+                'LEFT JOIN credit_creditcourse as cc ON cr.course_id=cc.id '
+                'LEFT JOIN credit_creditprovider as cp ON cr.provider_id=cp.id '
+                'WHERE cr.username=%s '
+                'ORDER BY cc.course_key ASC, cp.provider_id ASC '
+            ), [username]
+        )
+
+        return [
+            {
+                "uuid": row[0],
+                "timestamp": row[1],
+                "course_key": row[2],
+                "provider": {
+                    "id": row[3],
+                    "display_name": row[4]
+                },
+                "status": row[5]
+            }
+            for row in cursor.fetchall()
+        ]
+
+    class Meta(object):  # pylint: disable=missing-docstring
+        # Enforce the constraint that each user can have exactly one outstanding
+        # request to a given provider.  Multiple requests use the same UUID.
+        unique_together = ('username', 'course', 'provider')
+
+
+class CreditRequestStatus(TimeStampedModel):
```

Comment:
Minor: [django-simple-history](https://django-simple-history.readthedocs.org/en/latest/) would allow you to track all changes to `CreditRequest`, and remove the need for this model.